yarn-project/noir-contracts/src/contracts/ecdsa_account_contract/src/ecdsa_public_key_note.nr

```diff
@@ -53,11 +53,11 @@ impl EcdsaPublicKeyNote {
     }
 
     fn compute_nullifier(self) -> Field {
-        let siloed_note_hash = compute_siloed_note_hash(EcdsaPublicKeyNoteInterface, self);
+        let unique_siloed_note_hash = compute_unique_siloed_note_hash(EcdsaPublicKeyNoteInterface, self);
```

Comment:
Trying to understand / recall how the nullifier for a pending note is to be computed.
I'm wondering if we should use your `compute_note_hash_for_read_request` function here, instead of `compute_unique_siloed_note_hash`.

Otherwise, in the case of nullifying a pending note, we'll have:
- the note in the read request is `inner_note_hash`
- the note used in the nullifier (a 'pending_note_nullifier' ) is `h(0, inner_note_hash)` (assuming `header.nonce == 0` for all pending notes).

Does this discrepancy (`inner_note_hash` used in the read, but `h(0, inner_note_hash)` used in the nullifier) cause problems / lead to confusion?

Reply:
Also, the term 'unique siloed...' requires some detailed understanding of the protocol, adding friction before a dev can confidently write their own custom notes.

Reply:
I think you might be right that we should create nullifiers using the same note hash as is used for read requests - if the note being read or nullified is pending, use `inner_note_hash`, otherwise use `unique_siloed...`.

Agreed that it is confusing.... But I also don't want it to be unclear :thinking: We have a note hash, an inner note hash, a siloed note hash, and a unique siloed note hash.... We want it to be intuitive, but also we want it to be very clear what each is.

Reply:
I am renaming the function `compute_note_hash_for_read_or_nullify` and renaming the local variable `note_hash_for_nullify`